cloudinit/dmi.py

```diff
@@ -32,10 +32,32 @@
     'system-version': 'product_version',
 }
 
+# FreeBSD's kenv and dmidecode use different names for the same value,
+# this allows us to refer to them by one cannonical name
+DMIDECODE_TO_KENV_MAPPING = {
+    'baseboard-asset-tag': 'smbios.planar.tag',
+    'baseboard-manufacturer': 'smbios.planar.maker',
+    'baseboard-product-name': 'smbios.planar.product',
+    'baseboard-serial-number': 'smbios.planar.serial',
+    'baseboard-version': 'smbios.planar.version',
+    'bios-release-date': 'smbios.bios.reldate',
+    'bios-vendor': 'smbios.bios.vendor',
+    'bios-version': 'smbios.bios.version',
+    'chassis-asset-tag': 'smbios.chassis.tag',
+    'chassis-manufacturer': 'smbios.chassis.maker',
+    'chassis-serial-number': 'smbios.chassis.serial',
+    'chassis-version': 'smbios.chassis.version',
+    'system-manufacturer': 'smbios.system.maker',
+    'system-product-name': 'smbios.system.product',
+    'system-serial-number': 'smbios.system.serial',
+    'system-uuid': 'smbios.system.uuid',
+    'system-version': 'smbios.system.version',
+}
+
```

Comment:
You could potentially do something with a named tuple to avoid copying the copy/paste between keys in DMIDECODE_TO_KENV_MAPPING and DMIDECODE_TO_DMI_SYS_MAPPING.  Maybe that would make things easier and maybe it would be more obvious to someone adding a key that they had to think about dmidecode.

Here is the idea:

```python
from collections import namedtuple


kdmi= namedtuple('KernelNames', ['linux', 'freebsd'])
kdmi.__new__.defaults__ = (None, None)

DMIDECODE_TO_KERNEL = {
    'baseboard-asset-tag': kdmi('board_asset_tag', 'smbios.planar.tag'),
    'baseboard-manufacturer': kdmi('board_vendor', 'smbios.planar.maker'),
    'baseboard-product-name': kdmi('board_name', 'smbios.planar.product'),
    'baseboard-serial-number': kdmi('board_serial', 'smbios.planar.serial'),
}
DMI_SYS_PATH = "/sys/class/dmi/id"

def _read_dmi_syspath(key):
    kmap = DMIDECODE_TO_KERNEL.get(key)
    if kmap is None or kmap.linux is None:
        return None
    dmi_key_path = "{0}/{1}".format(DMI_SYS_PATH, kmap.linux)
    print("here i would read %s" % dmi_key_path)
    

def _read_kenv(key):
    kmap = DMIDECODE_TO_KERNEL.get(key)
    if kmap is None or kmap.freebsd is None:
        return None
    print("here i subp kenv -q %s" % kmap.freebsd)


_read_dmi_syspath("baseboard-manufacturer")
_read_kenv("baseboard-manufacturer")

```

Then running it shows:

```
here i would read /sys/class/dmi/id/board_vendor
here i subp kenv -q smbios.planar.maker
```

Reply:
i love this idea, and am looking forward to seeing your suggestion for doing this in posix sh for ds-identify ;)